backend/chainlit/playground/providers/vertexai.py

```diff
@@ -0,0 +1,117 @@
+from chainlit.playground.config import BaseProvider, add_llm_provider
+
+from fastapi.responses import StreamingResponse
+
+from chainlit.input_widget import Select, Slider, Tags
+from chainlit.playground.provider import BaseProvider
+
+vertexai_common_inputs = [
+    Slider(
+        id="temperature",
+        label="Temperature",
+        min=0.0,
+        max=0.99,
+        step=0.01,
+        initial=0.2,
+    ),
+    Slider(
+        id="max_output_tokens",
+        label="Max Output Tokens",
+        min=0.0,
+        max=1024,
+        step=1,
+        initial=256,
+    ),
+]
+
+
+class ChatVertexAI(BaseProvider):
+    async def create_completion(self, request):
+        await super().create_completion(request)
+        from vertexai.language_models import ChatModel, CodeChatModel
+
+        self.validate_env(request=request)
+
+        llm_settings = request.prompt.settings
+        self.require_settings(llm_settings)
+
+        prompt = self.create_prompt(request)
+        model_name = llm_settings["model"]
+        if model_name.startswith("chat-"):
+            model = ChatModel.from_pretrained(model_name)
+        elif model_name.startswith("codechat-"):
+            model = CodeChatModel.from_pretrained(model_name)
+        else:
+            raise ValueError(f"This model{model_name} is not implemented.")
+        del llm_settings["model"]
+        chat = model.start_chat()
+
+        def create_event_stream():
+            for response in chat.send_message_streaming(
+                prompt[0].formatted, **llm_settings
+            ):
+                yield response.text
+
+        return StreamingResponse(create_event_stream())
+
+
+class GenerationVertexAI(BaseProvider):
+    async def create_completion(self, request):
+        await super().create_completion(request)
+        from vertexai.language_models import TextGenerationModel, CodeGenerationModel
+
+        self.validate_env(request=request)
+
+        llm_settings = request.prompt.settings
+        self.require_settings(llm_settings)
+
+        prompt = self.create_prompt(request)
+        model_name = llm_settings["model"]
+        if model_name.startswith("text-"):
+            model = TextGenerationModel.from_pretrained(model_name)
+        elif model_name.startswith("code-"):
+            model = CodeGenerationModel.from_pretrained(model_name)
+        else:
+            raise ValueError(f"This model{model_name} is not implemented.")
+        del llm_settings["model"]
+
+        def create_event_stream():
+            for response in model.predict_streaming(prompt, **llm_settings):
```

Comment:
same comment as above for async / make_async

Reply:
Same as well.